processor/src/chiplets/mod.rs

```diff
@@ -29,7 +29,7 @@ use kernel_rom::KernelRom;
 mod bus;
 pub use bus::{AuxTraceBuilder, ChipletsBus};
 
-#[cfg(test)]
+#[cfg(all(test, feature = "internals"))]
```

Comment:
Shouldn't it be `any`? Meaning: It is going to be activated if either `test` "OR" `internals` are enabled. `all` will `test` "AND" `internals`

Reply:
The tests module has a dependency on `internals` so we will always need `internals` active when `test` is. For the other case we could compile the `tests`  if only `internals` is active `test` is not but I don't think that makes sense.

Reply:
Upon review I think you're right.  If I consistently mark as `any` it works as expected.  Good catch, I'll get this updated tomorrow.